tests/unit/template_cassdc_test.go

```diff
@@ -338,8 +343,8 @@ var _ = Describe("Verify CassandraDatacenter template", func() {
 			Expect(cassdc.Spec.PodTemplateSpec.Spec.Containers[0].Env).To(BeNil())
 			// No initcontainers slice should be present
 			Expect(cassdc.Spec.PodTemplateSpec.Spec.InitContainers).To(BeNil())
-			// No users should exist
-			Expect(cassdc.Spec.Users).To(BeNil())
+			// With Stargate enabled by default, only the stargate user should exist
+			Expect(cassdc.Spec.Users).To(ConsistOf(cassdcv1beta1.CassandraUser{Superuser: true, SecretName: clusterName + "-stargate"}))
```

Comment:
@burmanm I updated the test you modified in #357 as discussed over there.

Reply:
Ah, the test was designed to verify that empty users never exists (nil, not size 0), since that fails the CassandraDatacenter deployment. Other tests can verify that the correct user appears.

Reply:
Ah, OK, so then we actually want this one to disable stargate as well.

Reply:
OK, swapped it to disable stargate for the test and switched back the check to be the nil check.